src/packet.rs

```diff
@@ -211,31 +211,17 @@ impl Header {
             }
         };
 
-        let (dcil, scil) = match b.get_u8() {
-            Ok(v) => {
-                let mut dcil = v >> 4;
-                let mut scil = v & 0xf;
-
-                if dcil > 0 {
-                    dcil += 3;
-                }
-
-                if scil > 0 {
-                    scil += 3;
-                }
-
-                if dcil > MAX_CID_LEN || scil > MAX_CID_LEN {
-                    return Err(Error::InvalidPacket);
-                }
-
-                (dcil, scil)
-            },
-
-            Err(_) => return Err(Error::BufferTooShort),
-        };
+        let dcid_len = b.get_u8()?;
+        if version == crate::PROTOCOL_VERSION && dcid_len > 20 {
```

Comment:
Better use `MAX_CID_LEN` here instead of hardcoding the value.
```suggestion
        if version == crate::PROTOCOL_VERSION && dcid_len > MAX_CID_LEN {
```

Reply:
derp, thnx.